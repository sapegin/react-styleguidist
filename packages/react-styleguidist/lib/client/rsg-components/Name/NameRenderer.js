import "core-js/modules/es6.function.name";
import "core-js/modules/es6.string.small";

import React from 'react';
import PropTypes from 'prop-types';
import Code from 'rsg-components/Code';
import Styled from 'rsg-components/Styled';
import cx from 'classnames';

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
export var styles = function styles(_ref) {
  const fontSize = _ref.fontSize;

      
const color = _ref.color;
  return {
    name: {
      fontSize: fontSize.small,
      color: color.name
    },
    isDeprecated: {
      color: color.light,
      textDecoration: 'line-through'
    }
  };
};
export function NameRenderer(_ref2) {
  const classes = _ref2.classes;

      
const children = _ref2.children;

      
const deprecated = _ref2.deprecated;
  const classNames = cx(classes.name, _defineProperty({}, classes.isDeprecated, deprecated));
  return React.createElement("span", {
    className: classNames
  }, React.createElement(Code, null, children));
}
NameRenderer.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.node.isRequired,
  deprecated: PropTypes.bool
};
export default Styled(styles)(NameRenderer);