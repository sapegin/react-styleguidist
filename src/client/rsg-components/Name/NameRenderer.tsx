import React from 'react';
import cx from 'clsx';
import Styled, { JssInjectedProps, Theme } from 'rsg-components/Styled';

export const styles = ({ fontFamily, fontSize, color }: Theme) => ({
	name: {
		fontFamily: fontFamily.monospace,
		fontSize: fontSize.small,
		color: color.name,
	},
	isDeprecated: {
		color: color.light,
		textDecoration: 'line-through',
	},
});

interface NameProps extends JssInjectedProps {
	deprecated?: boolean;
}

export const NameRenderer: React.FunctionComponent<NameProps> = ({
	classes,
	children,
	deprecated,
}) => {
	const classNames = cx(classes.name, {
		[classes.isDeprecated]: deprecated,
	});
	return <code className={classNames}>{children}</code>;
};

export default Styled(styles)(NameRenderer);
