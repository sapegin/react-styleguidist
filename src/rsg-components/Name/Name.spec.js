import React from 'react';
import { NameRenderer } from './NameRenderer';

const classes = {
	name: 'name',
	isDeprecated: 'isDeprecated',
};

it('renderer should render argument name', () => {
	const actual = shallow(<NameRenderer classes={classes} name="Foo" />);

	expect(actual).toMatchSnapshot();
});

it('renderer should render deprecated argument name', () => {
	const actual = shallow(<NameRenderer classes={classes} name="Foo" deprecated />);

	expect(actual).toMatchSnapshot();
});
