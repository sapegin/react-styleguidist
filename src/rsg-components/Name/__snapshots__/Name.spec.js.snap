// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render argument name 1`] = `
<span
  className="name"
>
  <Consumer(Code)>
    Foo
  </Consumer(Code)>
</span>
`;

exports[`renderer should render deprecated argument name 1`] = `
<span
  className="name isDeprecated"
>
  <Consumer(Code)>
    Foo
  </Consumer(Code)>
</span>
`;
